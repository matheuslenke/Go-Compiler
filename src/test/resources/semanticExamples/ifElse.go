package samples

import fmt "fmt"

func testeIf() {

	var a int = 2

	var b bool = true
	var c bool = false

	if (a > 0) {
		fmt.Println("A é maior que 0")
		if (a < 0) {			
			fmt.Println("A é menor que 0")
		} else {
			fmt.Println("A é maior que 0")
		}
	} else {
		fmt.Println("A é menor que 0")
	}

	if (b == true) {
		fmt.Println("True")
	} 
	
	if (b == false) {
		fmt.Println("False")
	}

	if a <= 2 {
		fmt.Println("A é menor igual a 2")
	} else {
			fmt.Println("A é maior igual a 2")
	}

	if (a >= 5) {
		fmt.Println("A maior igual a 5")
	} else {
		fmt.Println("A menor que 5")
	}
}