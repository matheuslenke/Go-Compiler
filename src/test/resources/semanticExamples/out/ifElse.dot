digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) testeIf@5"];
node11[label="parameters"];
node10 -> node11;
node12[label="block"];
node13[label="var_list"];
node14[label="="];
node15[label="(int) a@6"];
node14 -> node15;
node16[label="(int) int@2"];
node14 -> node16;
node13 -> node14;
node12 -> node13;
node17[label="var_list"];
node18[label="="];
node19[label="(bool) b@7"];
node18 -> node19;
node20[label="(bool) bool@true"];
node18 -> node20;
node17 -> node18;
node12 -> node17;
node21[label="var_list"];
node22[label="="];
node23[label="(bool) c@8"];
node22 -> node23;
node24[label="(bool) bool@false"];
node22 -> node24;
node21 -> node22;
node12 -> node21;
node25[label="if"];
node26[label="(bool) >"];
node27[label="(int) a@6"];
node26 -> node27;
node28[label="(int) int@0"];
node26 -> node28;
node25 -> node26;
node29[label="block"];
node30[label="write"];
node31[label="Println@1"];
node30 -> node31;
node32[label="arguments"];
node33[label="(string) string@0"];
node32 -> node33;
node30 -> node32;
node29 -> node30;
node34[label="if"];
node35[label="(bool) <"];
node36[label="(int) a@6"];
node35 -> node36;
node37[label="(int) int@0"];
node35 -> node37;
node34 -> node35;
node38[label="block"];
node39[label="write"];
node40[label="Println@1"];
node39 -> node40;
node41[label="arguments"];
node42[label="(string) string@1"];
node41 -> node42;
node39 -> node41;
node38 -> node39;
node34 -> node38;
node43[label="block"];
node44[label="write"];
node45[label="Println@1"];
node44 -> node45;
node46[label="arguments"];
node47[label="(string) string@1"];
node46 -> node47;
node44 -> node46;
node43 -> node44;
node34 -> node43;
node29 -> node34;
node25 -> node29;
node48[label="block"];
node49[label="write"];
node50[label="Println@1"];
node49 -> node50;
node51[label="arguments"];
node52[label="(string) string@0"];
node51 -> node52;
node49 -> node51;
node48 -> node49;
node53[label="if"];
node54[label="(bool) <"];
node55[label="(int) a@6"];
node54 -> node55;
node56[label="(int) int@0"];
node54 -> node56;
node53 -> node54;
node57[label="block"];
node58[label="write"];
node59[label="Println@1"];
node58 -> node59;
node60[label="arguments"];
node61[label="(string) string@1"];
node60 -> node61;
node58 -> node60;
node57 -> node58;
node53 -> node57;
node62[label="block"];
node63[label="write"];
node64[label="Println@1"];
node63 -> node64;
node65[label="arguments"];
node66[label="(string) string@1"];
node65 -> node66;
node63 -> node65;
node62 -> node63;
node53 -> node62;
node48 -> node53;
node25 -> node48;
node12 -> node25;
node67[label="if"];
node68[label="(bool) =="];
node69[label="(bool) b@7"];
node68 -> node69;
node70[label="(bool) bool@true"];
node68 -> node70;
node67 -> node68;
node71[label="block"];
node72[label="write"];
node73[label="Println@1"];
node72 -> node73;
node74[label="arguments"];
node75[label="(string) string@2"];
node74 -> node75;
node72 -> node74;
node71 -> node72;
node67 -> node71;
node12 -> node67;
node76[label="if"];
node77[label="(bool) =="];
node78[label="(bool) b@7"];
node77 -> node78;
node79[label="(bool) bool@false"];
node77 -> node79;
node76 -> node77;
node80[label="block"];
node81[label="write"];
node82[label="Println@1"];
node81 -> node82;
node83[label="arguments"];
node84[label="(string) string@3"];
node83 -> node84;
node81 -> node83;
node80 -> node81;
node76 -> node80;
node12 -> node76;
node85[label="if"];
node86[label="(bool) <="];
node87[label="(int) a@6"];
node86 -> node87;
node88[label="(int) int@2"];
node86 -> node88;
node85 -> node86;
node89[label="block"];
node90[label="write"];
node91[label="Println@1"];
node90 -> node91;
node92[label="arguments"];
node93[label="(string) string@4"];
node92 -> node93;
node90 -> node92;
node89 -> node90;
node85 -> node89;
node94[label="block"];
node95[label="write"];
node96[label="Println@1"];
node95 -> node96;
node97[label="arguments"];
node98[label="(string) string@4"];
node97 -> node98;
node95 -> node97;
node94 -> node95;
node85 -> node94;
node12 -> node85;
node99[label="if"];
node100[label="(bool) >="];
node101[label="(int) a@6"];
node100 -> node101;
node102[label="(int) int@5"];
node100 -> node102;
node99 -> node100;
node103[label="block"];
node104[label="write"];
node105[label="Println@1"];
node104 -> node105;
node106[label="arguments"];
node107[label="(string) string@5"];
node106 -> node107;
node104 -> node106;
node103 -> node104;
node99 -> node103;
node108[label="block"];
node109[label="write"];
node110[label="Println@1"];
node109 -> node110;
node111[label="arguments"];
node112[label="(string) string@5"];
node111 -> node112;
node109 -> node111;
node108 -> node109;
node99 -> node108;
node12 -> node99;
node10 -> node12;
node0 -> node10;
}
