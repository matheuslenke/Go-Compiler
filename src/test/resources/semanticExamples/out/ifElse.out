[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: A é maior que 0
Entry 1: True
Entry 2: False
Entry 3: A é menor igual a 2
Entry 4: A maior igual a 5




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: testeIf, line: 5, type: func, scopeName: global
Entry 6 -- name: a, line: 7, type: int, scopeName: testeIf
Entry 7 -- name: b, line: 9, type: bool, scopeName: testeIf
Entry 8 -- name: c, line: 10, type: bool, scopeName: testeIf




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.953 s
[INFO] Finished at: 2022-02-24T20:53:52-03:00
[INFO] ------------------------------------------------------------------------
